atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/TableDefinition.java

```diff
@@ -176,12 +176,26 @@ private void checkUniqueColumnNames(String columnName, String shortName) {
      */
     public void hashFirstRowComponent() {
         Preconditions.checkState(state == State.DEFINING_ROW_NAME,
-                "Can only indicate hashFirstRowComponent() inside the rowName scope.");
-        Preconditions.checkState(rowNameComponents.isEmpty(), "hashRowComponent must be the first row component");
-        hashFirstRowComponent = true;
+                "Can only indicate hashFirstRowComponents() inside the rowName scope.");
+        Preconditions.checkState(rowNameComponents.isEmpty(),
+                "hashFirstRowComponent must be the first row component");
+        hashFirstNRowComponents(1);
+    }
+
+    /**
+     * Prefix the row with a hash of the first N row components.
+     * If using prefix range requests, the components that are hashed must also be specified in the prefix.
+     */
+    public void hashFirstNRowComponents(int numberOfComponents) {
+        Preconditions.checkState(state == State.DEFINING_ROW_NAME,
```

Comment:
Extract the preconditions to a method `checkHashRowComponentsPreconditions(String methodname)`